src/lsp/cobol_parser/expect/gen_expect.ml

```diff
@@ -167,234 +175,238 @@ module CompEntrySet = Set.Make(struct
+          | _ -> None end
+        (Lr1.transitions lr1) in
     CompEntrySet.from_list_map
       ~init:custom_comp_entries ~add:CompEntrySet.add_list
       next_tokens_from_item (Lr0.items @@ Lr1.lr0 lr1)
   end
 
 let emit_acceptable_terminals_in_env ppf = (* taking transitions *)
-  Fmt.pf ppf {|
+  Fmt.string ppf {|
 let acceptable_terminals_in ~env : completion_entry list =
   NEL.(match Menhir.current_state_number env with
 |};
   Lr1.fold (fun lr1 acc ->
-    match completion_entries_of lr1 with
+      match completion_entries_of lr1 with
       | s when CompEntrySet.is_empty s -> acc
       | s -> CompEntryMap.add_to_list s lr1 acc)
-  CompEntryMap.empty
+    CompEntryMap.empty
   |> inv (CompEntryMap.fold (fun s l acc -> (l, CompEntrySet.elements s)::acc)) []
   |> pp_match_cases ppf
     Fmt.(using Lr1.to_int int)
     (pp_brackets @@ pp_list pp_completion_entry)
     "[])"
 
 let guess_default_value ~mangled_name ~typ =
-  match typ, mangled_name with
-  | _ when EzString.ends_with ~suffix:"option" typ ||
-          EzString.starts_with ~prefix:"option_" mangled_name
-  -> Some "None"
-  | _ when EzString.ends_with ~suffix:"list" typ ||
-  EzString.starts_with ~prefix:"loption_" mangled_name ||
-  EzString.starts_with ~prefix:"list_" mangled_name
-  -> Some "[]"
-  | _ when EzString.ends_with ~suffix:"unit" typ
-  -> Some "()"
-  | _ when EzString.ends_with ~suffix:"bool" typ ||
-  EzString.starts_with ~prefix:"boption_" mangled_name
-  -> Some "false"
-  | _ -> None
+  if EzString.ends_with ~suffix:" option" typ ||
+     EzString.starts_with ~prefix:"option_" mangled_name
+  then Some "None"
+  else if EzString.ends_with ~suffix:" list" typ ||
+          EzString.starts_with ~prefix:"loption_" mangled_name ||
+          EzString.starts_with ~prefix:"list_" mangled_name
+  then Some "[]"
+  else if String.equal typ "unit"
+  then Some "()"
+  else if String.equal typ "bool" ||
+          EzString.starts_with ~prefix:"boption_" mangled_name
+  then Some "false"
+  else None
 
 let is_nullable_without_recovery = Nonterminal.tabulate begin fun nt ->
-  Nonterminal.nullable nt &&
+    Nonterminal.nullable nt &&
     Nonterminal.attributes nt
     |> List.find_opt (Attribute.has_label "recovery")
     |> Option.is_none
   end
 
 let best_guess_default_value = Nonterminal.tabulate begin fun nt ->
-  if not (is_nullable_without_recovery nt)
-  then None
-  else
-  let mangled_name = Nonterminal.mangled_name nt in
-  let guessed_default = Option.bind
-    (Nonterminal.typ nt)
-    (fun typ -> guess_default_value ~mangled_name ~typ) in
-  let given_default = Nonterminal.attributes nt
-    |> List.find_opt (Attribute.has_label "default")
-    |> Option.map Attribute.payload in
-  match given_default with
-    | Some _ -> given_default
-    | None -> guessed_default
+    if not (is_nullable_without_recovery nt)
+    then None
+    else
+      let mangled_name = Nonterminal.mangled_name nt in
+      let guessed_default = Option.bind
```

Comment:
Not sure I understand what you mean

Reply:
The binding to `guessed_default` (and maybe `mangled_name`) is not used in case the `match` below returns `Some _`. So you can add the `Option.bind …` directly in the `None` case.